webrtc.html

```diff
@@ -3839,18 +3844,60 @@
+            <h2>Constructor</h2>
             <dl data-link-for="RTCIceCandidate" data-dfn-for="RTCIceCandidate"
             class="constructors">
               <dt><code>RTCIceCandidate</code></dt>
               <dd>
-                The <dfn><code>RTCIceCandidate()</code></dfn> constructor takes
+                <p>The <dfn><code>RTCIceCandidate()</code></dfn> constructor takes
                 a dictionary argument, <var>candidateInitDict</var>, whose
                 content is used to initialize the new <a class=
-                "idlType"><code>RTCIceCandidate</code></a> object. When run, if
-                both the <code><a>sdpMid</a></code> and
-                <code><a>sdpMLineIndex</a></code> dictionary members are
-                <code>null</code>, <a>throw</a> a <code>TypeError</code>.
+                "idlType"><code>RTCIceCandidate</code></a> object.</p>
+                <p>When invoked, run the following steps:</p>
+                <ol>
+                  <li>If both the <code>
+                  <a data-dfn-for="idl-def-rtcicecandidateinit-sdpmid">sdpMid</a></code> and <code>
+                  <a data-dfn-for="idl-def-rtcicecandidateinit-sdpmlineindex">sdpMLineIndex</a>
+                  </code> dictionary members in <var>candidateInitDict</var> are
+                  <code>null</code>, <a>throw</a> a <code>TypeError</code>.</li>
+                  <li>Let <var>iceCandidate</var> be a newly created
+                  <code>RTCIceCandidate</code> object.</li>
+                  <li>Set the corresponding attributes in <var>iceCandidate</var>
+                  with the dictionary member values of <var>candidateInitDict</var>.</li>
+                  <li>Let <var>candidate</var> be the <code>
+                  <a data-dfn-for="idl-def-rtcicecandidateinit-candidate">candidate</a></code>
+                  dictionary member of <var>candidateInitDict</var>. If
+                  <var>candidate</var> is not an empty string, run the following steps:
+                  <ol>
+                    <li>Parse <var>candidate</var> using the <a><code>candidate-attribute</code>
+                    </a> grammar.</li>
+                    <li>If parsing of <a><code>candidate-attribute</code></a> has failed, abort
+                    these steps.</li>
+                    <li>If any field in the parse result represents an invalid value for the
+                    corresponding attribute in <var>iceCandidate</var>, abort these steps.</li>
+                    <li>Set the corresponding attributes in <var>iceCandidate</var> to
+                    the field values of the parsed result.</li>
+                  </ol></li>
+                  <li>Return <var>iceCandidate</var>.</li>
+                </ol>
+                <div class="note">
+                  <p>The constructor for <code>RTCIceCandidate</code> only does basic
+                  type checking for the dictionary members in <var>candidateInitDict</var>.
```

Comment:
This sentence seems slightly misleading, since it doesn't just do basic type checking; it also does basic parsing.